sunflower-land/sunflower-land
# [FIX] State machine transaction order change. Fix error handling for create farm.

# Description

This PR moves the sign step to pre farm creation. The new state machine is now:
<img width="542" alt="Screen Shot 2022-02-07 at 8 15 48 pm" src="https://user-images.githubusercontent.com/17863697/152759353-dd15205d-217f-4675-bb2e-9058da917851.png">

This PR also fixes the error handling when the sign request fails. An error is now thrown so the state machine can catch it in the `onError` transition.

**NOTE: ** If there is a server error the user is taken to the "catch all" unauthorised state. This isn't a great user experience but we will discuss errors and the way to handle each of them separately.

Fixes # no issue

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

_Test transaction order_
- Create a new account in metamask
- Load the game
- Click on create farm and confirm that the `sign` transaction happens before the `createFarm` transaction

_Test server error_
- Create a new account in metamask
- Load the game
- Click on create farm and fail the api request
- You should be shown to the `Catch all errors` modal

# Checklist:
